### DISCUSSION TITLE
make ios scrcpy run in linux

### DISCUSSION THREAD

**Discussion by wanliLiu** [NetrisTV/ws-scrcpy#194] [Ideas]
currently ios device detect use lib `ios-device-lib`, and this lib can only run in mac or windows,not run in Linux. as for run xctest use appium default way `xcodebuild`, so , all of this lead to cannot run in linux.

and this lib [tidevice](https://github.com/alibaba/taobao-iphone-device) can fix it, use `tidevice list ` fix ios device detect, and  run xctest use `tidevice xctest -B com.facebook.wda.WebDriverAgent.Runner`, thus ,because it wirted in python,  so **it** can run in mac windows and linux.  I have done all of this in this project,it works,

by the way, video stream use mjpeg

**Comment by drauggres** [comment]
https://github.com/NetrisTV/ws-scrcpy/issues/195
